check_task.go

```diff
@@ -1,13 +1,15 @@
 package gosundheit
 
 import (
+	"context"
 	"time"
 )
 
 type checkTask struct {
 	stopChan chan bool
 	ticker   *time.Ticker
 	check    Check
+	timeout  time.Duration
```

Comment:
I think you should copy the context here upon task creation, rather than passing it to `execute()` every time. The execute is the correct place to add the timeout to the ctx

Reply:
The context is re-created for every execution because every execution has a new timeout